src/js/actions/documents.js

```diff
@@ -465,17 +465,16 @@ define(function (require, exports) {
                     recentFilesPromise = this.transfer(application.updateRecentFiles),
                     updateTransformPromise = this.transfer(ui.updateTransform),
                     deleteTempFilesPromise = this.transfer(libraryActions.deleteGraphicTempFiles,
-                        documentID, isDocumentSaved);
+                        documentID, isDocumentSaved),
+                    policyPromise = newDocument ? Promise.resolve() :
+                        this.transfer(toolActions.resetBorderPolicies);
 
                 return Promise.join(resetLinkedPromise,
                         resetHistoryPromise,
                         updateTransformPromise,
                         recentFilesPromise,
-                        deleteTempFilesPromise);
-            })
-            .then(function () {
-                // This must run after the above transfers. See #2785.
-                return this.transfer(toolActions.resetBorderPolicies);
+                        deleteTempFilesPromise,
+                        policyPromise);
```

Comment:
I saw the `policyPromise` has the comment `This must run after the above transfers`, so is it safe to put it within the other promises to run concurrently?